addons/account/models/account_move.py

```diff
@@ -3166,6 +3166,9 @@ def button_draft(self):
             move.mapped('line_ids.analytic_line_ids').unlink()
 
         self.mapped('line_ids').remove_move_reconcile()
+        for move in self:
```

Comment:
unlink in one time
`self.payment_id.paired_internal_transfer_payment_id.unlink()`

Reply:
@abla001 Will it work when posted as well?

Reply:
@jco-odoo this action is called when resetting to draft which means it was posted before

Reply:
@abla001 But it means the linked one is still posted and I wonder if it is possible to delete it at that moment.  It is also a bit of change of behaviour by deleting it suddenly.